src/pages/People.tsx

```diff
@@ -0,0 +1,55 @@
+/*eslint-disable*/
+
+import { useEffect, useState } from 'react';
+
+import { Loader, PeopleTable } from '../components';
+
+import { Person } from '../types';
+
+import { getPeople } from '../api';
+
+import { getPeopleWithParents } from '../helpers/getPeopleWithParents';
+
+export const People = () => {
+  const [hasErrors, setHasErrors] = useState(false);
+
+  const [people, setPeople] = useState<Person[] | null>(null);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(data => {
+        setPeople(getPeopleWithParents(data));
+      })
+      .catch(() => {
+        setHasErrors(true);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="box table-container">
+          {hasErrors ? (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              Something went wrong
+            </p>
+          ) : !isLoading ? (
+            people?.length! > 0 ? (
+              <PeopleTable people={people} />
+            ) : (
+              <p data-cy="noPeopleMessage">There are no people on the server</p>
+            )
+          ) : (
+            <Loader />
+          )}
```

Comment:
This condition looks quite complex and confusing, try breaking it down into several different conditions

Reply:
Not fixed

Reply:
Not fixed